lib/Avatar/AvatarWithPopover.js

```diff
@@ -16,57 +16,106 @@ import { Avatar, Icon } from '../';
  * </AvatarWithPopover>
  */
 class AvatarWithPopover extends Component {
+  constructor(props) {
+    super(props);
+    this.togglePopover = this.togglePopover.bind(this);
+    this.hidePopover = this.hidePopover.bind(this);
+    this.showPopover = this.showPopover.bind(this);
+  }
   state = {
     showing: false
   };
 
-  render() {
-    const popover = (
-      <Popover id={this.props.email} className={this.props.popoverClass}>
-        {this.props.children}
-      </Popover>
-    );
+  togglePopover() {
+    this.setState({ showing: !this.state.showing });
+  }
 
-    const onEnter = () => {
-      this.setState({ showing: true });
-    };
-    const onExited = () => {
-      this.setState({ showing: false });
-    };
+  hidePopover() {
+    this.setState({ showing: false });
+  }
 
+  showPopover() {
+    this.setState({ showing: true });
+  }
+
+  renderTrigger() {
     const classes = cx('button button--icon-only avatar__wrapper', {
       'is-active': this.state.showing
     });
 
+    const props = {};
+    if (typeof this.props.triggerEvent === 'string') {
```

Comment:
Maybe this could do with being abstracted into a method? Something like the following would be quite readable;

```
const triggerEventProps = getOverlayTriggerEventProps()
```

That object could then be spread below too e.g. 

```
<button className={classes} {...triggerEventProps}>
```